src/transform-sdk/rust/sr-sys/src/lib.rs

```diff
@@ -33,6 +33,7 @@ use redpanda_transform_sdk_sr_types::*;
 use redpanda_transform_sdk_varint as varint;
 
 use lru::LruCache;
+use std::cell::UnsafeCell;
```

Comment:
ace 🎯

Reply:
Let's use RefCell here please, I don't think the safety checks are a performance concern (and let's prove it with benchmarks first).

Reply:
Sure. I will note that I chose unsafecell because I was not aware refcell could also do this (i.e. not for perf reasons). Easy benchmark tho.

Reply:
Gotcha, no need to benchmark, let's just stick with RefCell for now

Reply:
This wound up kinda gross. `get` operations on the cache return optional borrows, but as I understand it those are bound to the lifetime of the `RefMut` that we peel off from `RefCell`, so you can't (?) return them (easily?). 

I suspect there's a way around it with `Ref::map` & friends, but then there's other weird stuff with passing in a `RefMut` to get a `Ref` out, etc.

Took a copy for now, but open to feedback.

Reply:
>kinda gross

or totally fine. fact remains that I didn't find a common pattern for this exact usage, but we've not introduced any additional clones afaict: https://github.com/redpanda-data/redpanda/pull/19859#discussion_r1676513624